target/linux/lantiq/dts/RS353.dts

```diff
@@ -0,0 +1,318 @@
+/dts-v1/;
+
+/include/ "vr9.dtsi"
+
+/ {
+	model = "RS353 - Bintec Elmeg RS353";
+
+	chosen {
+		bootargs = "console=ttyLTQ0,115200 init=/etc/preinit";
+		leds {
+                        running = &status;
+                        boot = &status;
+			wifi = &wlan;
+			usb2 = &usb; /* Front port is mapped to device 2 */
+			dsl = &dsl;
+                };
+
+	};
+
+	memory@0 {
+		reg = <0x0 0x8000000>;
+	};
+
+	sram@1F000000 {
+		cgu0: cgu@103000 {
+                        compatible = "lantiq,cgu-xway";
+                        reg = <0x103000 0x1000>;
+                        #clock-cells = <1>;
+			lantiq,phy-clk-src = <0x2>;
+		};
+	};
+
+	fpi@10000000 {
+		localbus@0 {
+			nor-boot@0 {
+				compatible = "lantiq,nor";
+				bank-width = <2>;
+				reg = <0 0x0 0x2000000>;
+				#address-cells = <1>;
+				#size-cells = <1>;
+
+				partitions {
+					compatible = "fixed-partitions";
+					#address-cells = <1>;
+					#size-cells = <1>;
+
+					boardconfig: partition@0 {
```

Comment:
I have to admit, that you got to a point where I might not be helpful anymore. I mean, I see what's the problem. But I fail to understand why this endianness swapping during nor probing isn't required in your case and what it means for PCI support. Same applies to borken early bootlogs messages. I do refrain from giving advices for issues I do not fully understand. But I will try to ask someone else for help. Might take a while, depending on his interest and workload.

But I can give you some more feedback regarding the remaining issues. First of all you need to adjust the partition table. I can see that the u-boot partitions still exists.

Regarding the boss header, it would be the best to add a tool to tools/firmware-utils which creates the required header and a custom mtd splitter to target/linux/generic/files/drivers/mtd/mtdsplit/. I don't like the idea of replacing the Bootmonitor with a patched version if there are other options.

Aynway, the checksum from the header will be invalid right after:

>```
>[    9.656152] jffs2_scan_eraseblock(): End of filesystem marker found at 0x20000
>[    9.662050] jffs2_build_filesystem(): unlocking the mtd device... done.
>[    9.668540] jffs2_build_filesystem(): erasing all blocks after the end marker... 
>```

pad-rootfs adds the 0xDEADCODE (?) magic (end of filesystem marker) to the end of the image. Due to an OpenWrt patch (sorry, to lazy to search which one it is) the system checks the flash for the magic during firstboot. If the magic is found, the flash portion beginning with the magic till the end of the firmware partition will jffs2 formatted.  This will be the rootfs_data partition. Since the 0xDEADCODE magic is gone after that, the checksum will not longer match.

I've two possible workarounds for the crc problem in mind:

1. find a way to flash an image bigger than the length field in the header
2. change the length header field + checksum header field on firstboot (maybe to lzmaloader+kernel+rootfs without 0xDEADCODE magic).

Something similar to 2. is already done for other boards. For example in [brcm63xx](https://github.com/openwrt/openwrt/blob/master/target/linux/brcm63xx/base-files/etc/uci-defaults/09_fix_crc). I'm quite sure there are more examples in other targets.

Another way of fixing the header could be to add the required code to your mtd splitter. Albeit using an uci-defaults script has the advantage that the script is removed after it runs, which ensures that the script runs only on firstboot after a new image was installed and doesn't waste cpu cycles for subsequently boots.

But something similar might be done using the mtd splitter as well. Just check if the end of the autosplitted rootfs partition is identical with the length from the header. If yes, do nothing. Otherwise adjust the values.